app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java

```diff
@@ -48,17 +48,24 @@ protected void onHandleIntent(Intent intent) {
         otherAlertSnooze =  Integer.parseInt(prefs.getString("other_alerts_snooze", "20"));
 
         long now = new Date().getTime();
+        
+        if (!bg_missed_alerts) {
+        	// we should not do anything in this case. Until there is a listener on UI changes, try again in 5 minutes
+            Log.d(TAG, "Setting timer to  5 minutes from now" );
+            checkBackAfterMissedTime(now + 5 * 60 *1000);
```

Comment:
the parameter of checkBackAfterMissedTime should actually be "in how much time should the check occur" , not "when"
If you call it with now + 5 * 6 * 1000, the alarm will actually be set to calendar.getTimeInMillis() + now + 5 * 6 * 1000

should be checkBackAfterMissedTime(5 * 60 *1000);